base/mpfr.jl

```diff
@@ -880,4 +880,18 @@ get_emin_max() = ccall((:mpfr_get_emin_max, :libmpfr), Clong, ())
 set_emax!(x) = ccall((:mpfr_set_emax, :libmpfr), Void, (Clong,), x)
 set_emin!(x) = ccall((:mpfr_set_emin, :libmpfr), Void, (Clong,), x)
 
+function Base.deepcopy_internal(x::BigFloat, stackdict::ObjectIdDict)
+    if haskey(stackdict, x)
+        return stackdict[x]
+    end
+    N = precision(x)
+    y = BigFloat(zero(Clong), zero(Cint), zero(Clong), C_NULL)
+    ccall((:mpfr_init2,:libmpfr), Void, (Ptr{BigFloat}, Clong), &y, N)
+    finalizer(y, Base.GMP._mpfr_clear_func)
```

Comment:
I'm not familiar with mpfr, but any reason not to replace those 4 lines by `y = setprecision(precision(x)) do; BigFloat() end` ?